src/System.Private.Xml/tests/Xslt/XslTransformApi/CXslTransform.cs

```diff
@@ -2928,6 +2747,13 @@ public void var4(InputType inputType, ReaderType readerType, TransformType trans
         }
     }
 
+    internal sealed class AllowDefaultResolverContext : IDisposable
```

Comment:
I believe `[assembly: CollectionBehavior(DisableTestParallelization = true)]` but there might be something better. I think manual lock might be better since you already implement dispose:
```csharp
var o = new object();
Monitor.Enter(o);
Monitor.Exit(o);
```